pages/s/[id].tsx

```diff
@@ -0,0 +1,33 @@
+import React, { ReactElement } from 'react';
+import { useRouter } from 'next/router';
+import { getRequest } from '../../src/utils/apiRequests/api';
+
+interface Props {}
+
+export default function DirectGift({}: Props): ReactElement {
+  const router = useRouter();
+  //   const [profile, setProfile] = React.useState(null);
+  React.useEffect(() => {
+    if (router.query.id !== undefined) {
+      async function loadPublicUserData() {
+        const newProfile = await getRequest(
+          `/public/v1.0/en/treecounter/${router.query.id}`
```

Comment:
Should use a new profile api; also fetch the api using the aggregate function;

And add tenant key and session id.

Reply:
This is already using tenant key and session id

We cannot use new profile API it is not yet ready